packages/core/src/providers/github.ts

```diff
@@ -68,6 +68,13 @@ export interface GitHubProfile {
   [claim: string]: unknown
 }
 
+/**
+ * You can customize GitHub's baseUrl to your own GitHub Enterprise Server.
+ */
```

Comment:
Putting the comment here will make our Typedoc plugin generates all the properties of `OAuthUserConfig` in the Github provider doc page 😅 which is not very nice. To remedy this issue, I suggest we do the following:
```ts

/**
 * Configuration for usage with GitHub Enterprise Server.
 */
export interface GithubEnterpriseConfig { 
	/**
	 * Customize our own GitHub Enterprise Server.
	 */
	baseUrl?: string
 }
```

below:
```ts
  config: OAuthUserConfig<GitHubProfile> & GithubEnterpriseConfig
```

The JSDoc for the interface is necessary for the type to be generated, and the JSDoc for the property is to explain it.

Reply:
I made the changes as suggested. I understand the intent, but manipulating types for it feels a bit weird.

A more semantically accurate definition would be like
```ts
interface GitHubCloudConfig extends OAuthUserConfig<GitHubProfile> { ... }

interface GitHubEnterpriseServerConfig extends OAuthUserConfig<GitHubProfile> { ... }

// then
GitHubCloudConfig | GitHubEnterpriseServerConfig
```